src/atldld/requests.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (C) 2021 EPFL/Blue Brain Project
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""REST API requests to the AIBS servers.
+
+More info: https://help.brain-map.org/display/api/Allen+Brain+Atlas+API
+"""
+import logging
+from dataclasses import dataclass, replace
+from typing import Any, Dict, Optional, Sequence, Tuple
+
+import requests
+
+logger = logging.getLogger(__name__)
+
+_API_BASE_URL = "https://api.brain-map.org/api/v2"
+
+
+class RMAError(Exception):
+    """Raised when the RMA response status is False."""
+
+
+@dataclass
+class RMAParameters:
+    """Abstraction of RMA parameters."""
+
+    model: str
+    criteria: Optional[Dict[str, Any]] = None
+    include: Optional[Sequence[str]] = None
+    start_row: Optional[int] = None
+    num_rows: Optional[int] = None
+
+    def __str__(self) -> str:
+        """Convert RMA parameters to URL parameters."""
+        flags = [f"model::{self.model}"]
+
+        # Criteria
+        if self.criteria is not None:
+            flags.append("rma::criteria")
+            flags.append("".join(f"[{k}$eq{v}]" for k, v in self.criteria.items()))
+
+        # Include
+        if self.include is not None:
+            flags.append("rma::include")
+            flags.extend(self.include)
+
+        # Options
+        if self.start_row is not None or self.num_rows is not None:
+            flag = "rma::options"
+            if self.start_row is not None:
+                flag += f"[start_row$eq{self.start_row}]"
+            if self.num_rows is not None:
+                flag += f"[num_rows$eq{self.num_rows}]"
+            flags.append(flag)
+
+        return f'criteria={",".join(flags)}'
+
+
+def rma_all(rma_parameters: RMAParameters) -> list:
+    """Send one or multiple RMA requests to get all data for given parameters.
+
+    Parameters
+    ----------
+    rma_parameters
+        The RMA query parameters. Corresponds to the part of the query URL
+        that follows the "?" character. The fields start_row and num_rows
+        must be equal to None because they are used/changed in this function
+
+    Returns
+    -------
+    msg : list
+        The data received from the server.
+    """
+    # Make a copy of the parameters (since we'll modify them) and at the same
+    # time set start_row and num_row. The value of 25k rows per request is
+    # the max on the AIBS side.
```

Comment:
Interesting. So I guess that is why you wrote this function, right? Otherwise there would be no other reason?

Reply:
I just noticed you answered this in the PR description:D haha

Reply:
he he, yeah. To sum up, the default is 50, the max is 25k. Especially with section images it's easy to get over the default value of 50, so `rma_all` just does the iterating for you in the background.

Reply:
Makes sense:)

(Can be resolved)